dhcp/pxe.go

```diff
@@ -94,6 +94,13 @@ func SetupPXE(ctx context.Context, rep, req *dhcp4.Packet) bool {
 		dhcplog.With("mac", req.GetCHAddr(), "xid", req.GetXID()).Info("no client GUID provided")
 	}
 
+	// Wanted by PXE clients to learn of PXE servers, not required but responses with PXEClient ClassID are preferred.
```

Comment:
in `Figure 2-2 PXE Client Timeouts` and `Figure 2-3 PXE Client Response to DHCP Server Containing a Proxy DHCP Service` and a few other places in the [pxespec](http://www.pix.net/software/pxeboot/archive/pxespec.pdf) seem to indicate that a DHCP offer must have `PXEClient` in option 60. that would make this comment misleading.

Reply:
I added more info in the commit message which has an excerpt from the spec:

Section 2.2.2 of the Intel PXE Spec states:

> The purpose of the timeouts is to ensure the PXE client gives precedence to servers supplying
> “PXEClient” specific configuration tags. The PXE boot ROM must function as a normal DHCP boot
> ROM in the absence of a PXE specific response. However, the PXE boot ROM must wait for
> specified times to see if a PXE response is available before using a non-PXE configuration.

Note _precedence_ in the spec, this lines up with our real world experience with *all* our nics in production. We we *not* setting the "PXEClient" class id when the first dhcp message comes in where we respond with filename:$our-iPXE-build.

Reply:
I totally understand that the real-world experience shows this option might not be needed but I believe the pxe spec shows that the server must send this option and I think we should implement the pxe spec regardless. All I'm suggesting is removing "not required" in the comment.

Reply:
The spec does not *require* it though. Section 2.2.2 Timeout calls out:

> However, the PXE boot ROM must wait for specified times to see if a PXE response is available before using a non-PXE configuration

This is a SHOULD, not MUST.

I'll reword the comment though to drop a mention of required or not though.

Reply:
Thanks. 

Section 2.2.2 Timeout, is talking about the client waiting for PXE and non-PXE configurations. without PXEClient we would be in the class of non-PXE configuration. I don't think we want to implement a non-PXE configuration. especially since the pxe spec is about implementing a PXE configuration.

Reply:
I've updated the comment.

Reply:
thanks, DHCP server/client interactions are terribly undocumented. I did happen to find [this](https://www.ibm.com/docs/en/tpmfi/7.1.1.16?topic=configuration-configuring-dhcp-server). Talks a little about the logic for option 60 with `PXEClient`.

Reply:
Its also from this time where the spec is written but failing smart/helpful was seen as good. Which leads to the spec and implementations to differ a lot :/ .